File: 2303_calculate_amount_paid_in_taxes/Solution.cs
// LeetCode 2303 - Calculate Amount Paid in Taxes
// https://leetcode.com/problems/calculate-amount-paid-in-taxes/

public class Solution {
    public double CalculateTax(int[][] brackets, int income) {
        double ans = 0;
        int prev = 0;
        foreach (var b in brackets) {
            int upper = b[0], percent = b[1];
            if (income <= prev) break;
            int taxable = (income < upper) ? income - prev : upper - prev;
            ans += taxable * percent / 100.0;
            prev = upper;
        }
        return ans;
    }
}

File: 2303_calculate_amount_paid_in_taxes/solution.php
<?php
// LeetCode 2303 - Calculate Amount Paid in Taxes
// https://leetcode.com/problems/calculate-amount-paid-in-taxes/

class Solution {
    function calculateTax($brackets, $income) {
        $ans = 0.0;
        $prev = 0;
        foreach ($brackets as $b) {
            $upper = $b[0];
            $percent = $b[1];
            if ($income <= $prev) break;
            $taxable = ($income < $upper) ? $income - $prev : $upper - $prev;
            $ans += $taxable * $percent / 100.0;
            $prev = $upper;
        }
        return $ans;
    }
}
